app/controllers/collection_controller.rb

```diff
@@ -186,8 +186,15 @@ def show
           end
         end
 
-        if params[:work_search]
-          @works = @collection.search_works(params[:work_search]).includes(:work_statistic).paginate(page: params[:page], per_page: 10)
+        # Coming from work title/metadata search
+        if params[:search_attempt_id]
+          @search_attempt = SearchAttempt.find_by(id: params[:search_attempt_id])
+          if session[:search_attempt_id] != @search_attempt.id
+            session[:search_attempt_id] = @search_attempt.id
+          end
+          @query = @search_attempt.query
```

Comment:
Would it make sense to have the @search_attempt model perform the search itself, combining this line and the @colllection.search_works call into one method?  That might encapsulate (some) search logic into the `SearchAttempt` model.

Reply:
Yes, I feel like this makes it much cleaner! I moved all of the result finding logic to a `results` method in the `SearchAttempt` model, for all search types.